src/components/CreatePost/CreatePost.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useEffect, useState } from 'react';
+import Button from '../core/Button/Button';
+import classes from './CreatePost.module.scss';
+import { HiOutlineCamera } from 'react-icons/hi2';
+import PostCreate from 'src/model/PostCreate';
+import { createPost } from 'src/services/PostService';
+import { toast } from 'react-toastify';
+import UserResponse from 'src/model/UserResponse';
+
+const CreatePost = () => {
+  const [content, setContent] = useState('');
+  const [file, setFile] = useState<File | null>(null);
+  const [user, setUser] = useState<UserResponse>();
+
+  useEffect(() => {
+    const userString = localStorage.getItem('user');
+    if (userString) {
+      setUser(JSON.parse(userString));
+    }
+  }, []);
+
+  const handleContentChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setContent(e.target.value);
+  };
+
+  const handleFileChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (e.target.files && e.target.files.length > 0) {
+      const selectedFile = e.target.files[0];
+      setFile(selectedFile);
+    }
+  };
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (!user) return;
+
+    if (content.length >= 300) {
+      toast.error('Maximum of 300 characters for content reached!');
+      return;
+    } else if (content.length === 0) {
+      toast.error('Post must contain content!');
+      return;
+    }
+
+    const postData: PostCreate = {
+      content: content,
+      file: file,
+      authorId: user.id
+    };
+
+    createPost(postData)
```

Comment:
No need for variable here, you can do it like this `createPost({content, file, authorId: user.id})`